autodiff/forward/real/real.hpp

```diff
@@ -854,7 +854,7 @@ auto repr(const Real<N, T>& x)
 {
     std::stringstream ss;
     ss << "autodiff.real(";
-    for(auto i = 0; i <= N; ++i)
+    for(decltype(N) i = 0; i <= N; ++i)
```

Comment:
There are other "styles" possible here, e.g.
* `for(auto i = 0u; ...`
* `for(std::size_t i = 0; ...`

Which is the preferred style for autodiff?